js/modules/authVerifyUser.js

```diff
@@ -0,0 +1,39 @@
+import userData from "../database/userData.js";
+import { regEmail, regPassword } from "./regexPatterns.js";
+
+const verifyLoginCredentials = (email, password) => {
+  for (const key in userData) {
+    if (userData[key].email === email) {
+      if (userData[key].password === password) {
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+};
+
+const CheckEmailExist = (email) => {
+  for (const key in userData) {
+    if (userData[key].email === email) {
+      return true;
+    }
+  }
+  return false;
+};
```

Comment:
_생각해볼만한 부분_

이 `CheckEmailExist`랑 `verifyLoginCredentials`는 database에서 읽어와서 확인한다는 가정이 있군요.

database는 클라이언트(사용자 디바이스/브라우저)가 아닌, 외부 (가상/물리) 머신에 있는 서비스고, 그걸 가정하신거같아요. 다른 말로하면, 외부(네트워크)서비스에 요청을 통해 결과를 확인하는 동작이라고 가정할 수 있겠네요.

그럼 다른 검증함수와 달리, 외부서비스에 의존하는, 강한 특징을 가진 함수니까 별도 모듈(api.js, database/api.js 등)을 만들어서 관리해도 좋겠습니다

Reply:
이렇게 써 볼 수도 있겠네요.

가독성 면에서 좋은지 순서대로 읽어보면, "userData의 values 중에서  하나라도(some) 있나?. user email이 같은 사람이?" 로 읽힐 거같아요

```suggestion
const getIfUserExist = (email) => Object.values(userData).some(user=> user.email === email)
```

Reply:
추후의 과제가 어떤 방식으로 유저 정보를 가져올 지 몰라서 우선 이런 방식으로 개발한 것 같습니다. 주차가 진행되면서 외부서비스에 의존하는 기능을 개발하는 요구사항이 더 추가된다면 api라는 모듈을 만들어서 관리하는 편이 확실히 더 나을 것 같네요.

some이라는 메서드를 배웠음에도 막상 직접 기능을 구현할 때는 정작 사용할 아이디어가 떠오르지 않았네요... 좀더 유연한 사고를 해야겠습니다!